connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/outbound/operation/OperationInvoker.java

```diff
@@ -78,6 +81,25 @@ private Object resolveVariableValue(
     return value;
   }
 
+  private Object resolveHeaderValue(
+      ParameterDescriptor.Header<?> headerDescriptor, Map<String, String> headers) {
+    String rawValue = headers.get(headerDescriptor.getName());
+    Object value;
+    try {
+      value = objectMapper.convertValue(rawValue, headerDescriptor.getType());
+    } catch (Throwable e) {
+      throw new RuntimeException(e);
+    }
+    if (headerDescriptor.isRequired() && value == null) {
+      throw new ConnectorInputException(
+          "Required variable '" + headerDescriptor.getName() + "' is missing in the job headers.");
```

Comment:
I guess we should not use the word 'variable' in this error message to avoid any confusion between different value sources